### PR TITLE
feature: make the style of scrollbar consistent across devices

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

File: src/table/TableBodyWrapper.jsx [hunk 1]
```diff
@@ -28,6 +28,39 @@ const useStyles = makeStyles({
       },
     },
   },
+  hideScrollbar: {
+    // hide scrollbar
+    scrollbarWidth: 'none' /* Firefox */,
+    msOverflowStyle: 'none' /* IE 10+ */,
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:34]: Not supported, so not needed

File: src/table/TableBodyWrapper.jsx [hunk 2]
```diff
@@ -28,6 +28,39 @@ const useStyles = makeStyles({
...
+      backgroundColor: 'transparent',
+    },
+    '&&::-webkit-scrollbar-thumb': {
+      backgroundColor: 'transparent',
+    },
+  },
+  showScrollbar: {
+    // show scrollbar
+    scrollbarWidth: 'thin' /* Firefox */,
+    msOverflowStyle: 'none' /* IE 10+ */,
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:50]: same here

File: src/table/TableBodyWrapper.jsx [hunk 3]
```diff
@@ -57,18 +90,43 @@ const TableBodyWrapper = ({
...
   return (
-    <TableBody className={`${classes.body} ${classes.cellBase}`}>
+    <TableBody
+      className={`${classes.body} ${classes.cellBase} ${
+        // in windows, make all scrollbar button, track and thumb transparent
+        // navigator.appVersion.indexOf('Win') !== -1 &&
+        classes.hideScrollbar
+      }  ${
+        // navigator.appVersion.indexOf('Win') !== -1 &&
+        isScrollbar && classes.showScrollbar
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:127]: Comments here are just cluttering, I think the classes are self explanatory.

Comment (haxxmaxx) [reply src/table/TableBodyWrapper.jsx:127]: Would this work?
```suggestion
        ${isScrollbar ? classes.showScrollbar : classes.hideScrollbar}
```

File: src/table/TableBodyWrapper.jsx [hunk 4]
```diff
@@ -57,18 +90,43 @@ const TableBodyWrapper = ({
...
+    // if (navigator.appVersion.indexOf('Win') !== -1) {
+    const handleScroll = () => {
+      isScrollbar === false && setIsScrollbar(true);
+      // Clear our timeout throughout the scroll
+      window.clearTimeout(isScrolling);
+      // Set a timeout to run after scrolling ends
+      isScrolling = setTimeout(() => setIsScrollbar(false), 100);
+    };
+    window.addEventListener('scroll', handleScroll, true);
+    return () => window.removeEventListener('resize', handleScroll);
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:115]: should be scroll?

File: src/table/TableBodyWrapper.jsx [hunk 5]
```diff
@@ -57,18 +90,43 @@ const TableBodyWrapper = ({
...
+  const [isScrollbar, setIsScrollbar] = useState(false);
 
   useEffect(() => {
     selDispatch({ type: 'set-enabled', payload: { isEnabled: selectionsEnabled } });
     setColumnRenderers(getColumnRenderers);
   }, [selectionsEnabled, columns.length]);
 
+  // Setup isScrolling variable
+  let isScrolling;
+  // eslint-disable-next-line consistent-return
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:102]: Bad idea. When you add more stuff here that `addSelectionListeners` I think that those listeners wont unmount.

File: src/table/TableBodyWrapper.jsx [hunk 6]
```diff
@@ -28,6 +28,39 @@ const useStyles = makeStyles({
       },
     },
   },
+  hideScrollbar: {
+    // hide scrollbar
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:32]: Self-explanatory

File: src/table/TableBodyWrapper.jsx [hunk 7]
```diff
@@ -28,6 +28,39 @@ const useStyles = makeStyles({
...
+    scrollbarWidth: 'none' /* Firefox */,
+    msOverflowStyle: 'none' /* IE 10+ */,
+    '&&::-webkit-scrollbar-track': {
+      webkitBoxShadow: 'none',
+      backgroundColor: 'transparent',
+    },
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:38]: I see that this, and some more css, is constant. Try to add it to the theme :)

File: src/table/TableBodyWrapper.jsx [hunk 8]
```diff
@@ -57,18 +90,43 @@ const TableBodyWrapper = ({
...
   useEffect(() => {
     addSelectionListeners(selectionsAPI, selDispatch, setShouldRefocus);
+    // to display scrolling bar when the user scrolls in windows
+    // if (navigator.appVersion.indexOf('Win') !== -1) {
+    const handleScroll = () => {
+      isScrollbar === false && setIsScrollbar(true);
+      // Clear our timeout throughout the scroll
+      window.clearTimeout(isScrolling);
+      // Set a timeout to run after scrolling ends
+      isScrolling = setTimeout(() => setIsScrollbar(false), 100);
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:112]: Is the timeout needed?

File: src/table/TableBodyWrapper.jsx (hunk 1 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx:34]: Okay, removed

File: src/table/TableBodyWrapper.jsx (hunk 2 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx:50]: Okay, removed

File: src/table/TableBodyWrapper.jsx (hunk 3 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx:127]: good. updated

File: src/table/TableBodyWrapper.jsx (hunk 4 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx]: right. Fixed

File: src/table/TableBodyWrapper.jsx (hunk 6 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx]: yes, removed

File: src/table/TableBodyWrapper.jsx (hunk 5 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx:97]: Looking into it. Not sure how to fix it now.

File: src/table/TableBodyWrapper.jsx (hunk 8 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx]: I just feel it is better to have a timeout to disappear the scrollbar

File: src/table/TableBodyWrapper.jsx (hunk 5 above)
Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx]: fixed

File: src/table/TableBodyWrapper.jsx (hunk 8 above)
Comment (haxxmaxx) [reply src/table/TableBodyWrapper.jsx:104]: hmm yes I agree

File: src/table/TableBodyWrapper.jsx (hunk 7 above)
Comment (haxxmaxx) [reply src/table/TableBodyWrapper.jsx:36]: ok then this is fine

File: src/table/TableBodyWrapper.jsx [hunk 9]
```diff
@@ -65,10 +108,21 @@ const TableBodyWrapper = ({
...
     addSelectionListeners(selectionsAPI, selDispatch, setShouldRefocus);
+    const bodyElement = document.getElementsByTagName('tbody')[0];
+    const showScroll = () => setIsScrollbar(true);
+    const hideScroll = () => setIsScrollbar(false);
+    bodyElement.addEventListener('mouseover', showScroll, true);
+    bodyElement.addEventListener('mouseleave', hideScroll, true);
+    return () => {
+      window.removeEventListener('mouseover', showScroll);
+      window.removeEventListener('mouseleave', hideScroll);
+    };
```
Comment (haxxmaxx) [inline src/table/TableBodyWrapper.jsx:119]: Isn't this the same thing as just using the hover pseudo-class?

Comment (LiKang6688) [reply src/table/TableBodyWrapper.jsx:119]: when using the hover pseudo-class, can we change the style of the scrollbar? Look into that.

Comment (haxxmaxx) [reply src/table/TableBodyWrapper.jsx:119]: yeah maybe hard to set `&&::-webkit-scrollbar-track` etc

**Comment by haxxmaxx** [comment]
I just tried this out and seems to work:
```
 scrollbar: {
    scrollbarWidth: 'none' /* Firefox */,
    '&&::-webkit-scrollbar-track': {
      webkitBoxShadow: 'none',
      backgroundColor: 'transparent',
    },
    '&&::-webkit-scrollbar': {
      width: '3px',
      backgroundColor: 'transparent',
    },
    '&&::-webkit-scrollbar-thumb': {
      backgroundColor: 'transparent',
    },
    '&&:hover': {
      scrollbarWidth: 'thin' /* Firefox */,
      '&&::-webkit-scrollbar-track': {
        webkitBoxShadow: 'none',
        backgroundColor: 'transparent',
      },
      '&&::-webkit-scrollbar': {
        width: '6px' /* width of vertical scrollbar */,
        height: '6px' /* height of horizontal scrollbar */,
        backgroundColor: 'transparent',
      },
      '&&::-webkit-scrollbar-thumb': {
        backgroundColor: '#acacac',
      },
    },
  },
```
No need for listeners or anything

**Comment by LiKang6688** [comment]
> I just tried this out and seems to work:
> 
> ```
>  scrollbar: {
>     scrollbarWidth: 'none' /* Firefox */,
>     '&&::-webkit-scrollbar-track': {
>       webkitBoxShadow: 'none',
>       backgroundColor: 'transparent',
>     },
>     '&&::-webkit-scrollbar': {
>       width: '3px',
>       backgroundColor: 'transparent',
>     },
>     '&&::-webkit-scrollbar-thumb': {
>       backgroundColor: 'transparent',
>     },
>     '&&:hover': {
>       scrollbarWidth: 'thin' /* Firefox */,
>       '&&::-webkit-scrollbar-track': {
>         webkitBoxShadow: 'none',
>         backgroundColor: 'transparent',
>       },
>       '&&::-webkit-scrollbar': {
>         width: '6px' /* width of vertical scrollbar */,
>         height: '6px' /* height of horizontal scrollbar */,
>         backgroundColor: 'transparent',
>       },
>       '&&::-webkit-scrollbar-thumb': {
>         backgroundColor: '#acacac',
>       },
>     },
>   },
> ```
> 
> No need for listeners or anything. Also, add some extra pixels for the scrollbar, it is really small now. The macOS one actually gets bigger when dragging.

Nice. Updated

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**